src/containers/home.jsx

```diff
@@ -1,20 +1,73 @@
-import React from 'react';
+import React, { Component } from 'react';
 import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import slugify from 'slugify';
+
+import * as actions from '../store/actions/drinks';
 
 import Layout from '../misc/layout';
 
-const Home = () => (
-  <Layout className="home">
-    <h1>Home</h1>
-    <p>Lorem ipsum dolor sit amet.</p>
-    <div className="searcher">
-      <input placeholder="Search by name or ingredient" />
-      <button className="btn-send" type="button">
-        Search
-      </button>
-    </div>
-    <Link to="/categories" className="link">List categories</Link>
-  </Layout>
-);
-
-export default Home;
+class Home extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      needle: '',
+    };
+  }
+
+  setNeedle(e) {
```

Comment:
i would seriously reconsider this name, it makes sense when you explain it but not too much when anybody else look at your code, so if you don't want to change the name, would be better to give more context with a little comment